libfreerdp/core/rdp.c

```diff
@@ -1764,31 +1795,121 @@ int rdp_recv_callback(rdpTransport* transport, wStream* s, void* extra)
 
 			break;
 
-		case CONNECTION_STATE_CAPABILITIES_EXCHANGE:
+		case CONNECTION_STATE_MULTITRANSPORT_BOOTSTRAPPING_REQUEST:
+			if (!rdp_client_connect_auto_detect(rdp, s))
+			{
+				rdp_client_transition_to_state(
+				    rdp, CONNECTION_STATE_CAPABILITIES_EXCHANGE_DEMAND_ACTIVE);
+				status = -23;
+			}
+			break;
+
+		case CONNECTION_STATE_CAPABILITIES_EXCHANGE_DEMAND_ACTIVE:
 			status = rdp_client_connect_demand_active(rdp, s);
 
 			if (status < 0)
 				WLog_DBG(TAG,
 				         "%s: %s - "
 				         "rdp_client_connect_demand_active() - %i",
 				         __FUNCTION__, rdp_get_state_string(rdp), status);
-
+			else if (status == 1)
+				status = 1;
```

Comment:
This looks awkward - maybe just `break;` to make the code more clear?

Reply:
done